orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py

```diff
@@ -422,9 +434,19 @@ def permute_and_reshape_tensor(
     return new_tensor, shape_tensor
 
 
-@register_symbolic("einsum")
+@register_symbolic("einsum", torch_version_end="1.13.0")
 @parse_args("s", "v")
 def einsum(g, equation, tensor_list):
+    return einsum_internal(g, equation, tensor_list)
+
+
+@register_symbolic("einsum", torch_version_start="1.13.0")
+@parse_args("s", "v", "is")
+def einsum(g, equation, tensor_list, path=None):
```

Comment:
Does this work? python does not support function overloading. So not sure what the intention here is with defining the `einsum` function twice.

Reply:
PyTorch changed the signature of einsum from version 1.13 but we need to support both lower versions and higher versions. For lower versions, we need to register the one without path in the argument list, for higher, we need the new one. I tested this on 1.12 to 1.14 and it works. I think it's because for a given torch version, we will only register one of them to the custom op symbolic, so there is no conflict from exporter side. From our side, we never call these functions directly. I can try to rename this two functions to different names, if the custom op symbolic registry in exporter side can support a function name not same as the original name, it will be fine.

Reply:
ok, I think let's try to see if having different names works. If not, we can let it be this way.

Reply:
It works. Just rename the functions.